File: 003 Coding/Quiz.scala
import scala.collection.mutable.ListBuffer
object Quiz {
  def main(args: Array[String]): Unit = {
    var numList: ListBuffer[Int] = ListBuffer()
    for (x <- 0 until 30) {
      numList += scala.util.Random.nextInt()
    }

    def nameAndID(name: String, ID: String): String = {
      return name + ID
    }

    nameAndID("Carl", "123")

    def removeDuplicates(objectList: ListBuffer[String]) =
    {
      var varList : ListBuffer[String] = ListBuffer()
      for (x <- 0 until objectList.length)
      {
        for (j<- 1 until objectList.length)
          {
            if (objectList(j) == objectList(x))
            {
              objectList.patch(j, objectList, 1)
            }
          }
      }
      print(objectList)
    }

    var testList: ListBuffer[String] = ListBuffer("sun","mon","sun")
    removeDuplicates(testList)


    def smallestMissingInteger(numList: List[Int]) =
    {
      if (numList.min < 0)
      {
        var min: Int = numList.min
        while (min < 0)
        {
          min = min + 1
        }
        while (numList.contains(min))
        {
          min = min + 1
        }
        print(min)
      }
      if (numList.min > 0)
      {
        print(numList.min)
      }
    }
    var testNums:List[Int] = List(-2,-3,-1,0,1,2,5,6)
    smallestMissingInteger(testNums)

  }
}

File: 003_Coding/Quiz (Python)/Quiz.py
from random import random
for x  in range(30):
    list = [random]


def nameAndID (name, ID):
    return name + ID


def removeDuplicate(objList):
    for x in range(len(objList)):
        for y in range(x+1, len(objList)):
            if objList[x] == objList[y]:
                objList.pop(y)
                return objList


print(removeDuplicate(["sun","mon","sun"]))


def smallestMissingInteger(numList):
    if min(numList) < 0:
        value = min(numList)
        while value <= 0:
            value += 1
        while value in numList:
            value += 1
    return value

    if min(numList) > 0:
        return min(numList)


print(smallestMissingInteger([-1,-2,0,3,5]))
